tests/arc/topt_no_cursor.nim

```diff
@@ -61,9 +61,11 @@ var
 try:
   it_cursor = x
   a = (
-    :tmpD = `=dup`(it_cursor.key)
+    `=wasMoved`(:tmpD)
+    `=copy`(:tmpD, it_cursor.key)
```

Comment:
That's a nasty regression. Why can't it use =dup here?

Reply:
It's a tuple, which it doesn't look inside for now. Probably there is an object 
with a `=copy` but not a `=dup` inside a tuple, which might cause issues.

Reply:
Well it is

```nim
type
  KeyValue = tuple[key, val: seq[int]]

proc tt(x: KeyValue) =
  var it = x
  let a = (it.key, it.val)
  echo a
```

It could lift the `=dup` for this type as there is an explicit =dup for `seq`. Or is that missing?